.github/workflows/experimental.yml

```diff
@@ -0,0 +1,153 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Experimental Fuzzer Jobs"
+
+on:
+  pull_request:
+    paths:
+      - ".github/workflows/experimental.yml"
+
+  schedule:
+    - cron: '0 1 * * *'
+
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'Ref to checkout out'
+        default: 'main'
+      numThreads:
+        description: 'Number of threads'
+        default: 16
+      maxHighMemJobs:
+        description: 'Number of high memory jobs'
+        default: 8
+      maxLinkJobs:
+        description: 'Maximum number of link jobs'
+        default: 4
+      extraCMakeFlags:
+        description: 'Additional CMake flags'
+        default: ''
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+jobs:
+  compile:
+    runs-on: 8-core
+    timeout-minutes: 120
+    env:
+      CCACHE_DIR: "${{ github.workspace }}/.ccache/"
+      CCACHE_BASEDIR: "${{ github.workspace }}"
+      LINUX_DISTRO: "ubuntu"
+    steps:
+
+      - name: "Restore ccache"
+        uses: actions/cache@v3
+        with:
+          path: "${{ env.CCACHE_DIR }}"
+          # We are using the benchmark ccache as it has all
+          # required features enabled, so no need to create a new one
+          key: ccache-benchmark-${{ github.sha }}
+          restore-keys: |
+            ccache-benchmark-
+
+      - name: "Checkout Repo"
+        uses: actions/checkout@v3
+        with:
+          path: velox
+          submodules: 'recursive'
+          ref: "${{ inputs.ref || 'main' }}"
+
+      - name: "Install dependencies"
+        run: cd  velox && source ./scripts/setup-ubuntu.sh
+
+      - name: "Build"
+        run: |
+          cd velox
+          make debug NUM_THREADS="${{ inputs.numThreads || 8 }}" MAX_HIGH_MEM_JOBS="${{ inputs.maxHighMemJobs || 8 }}" MAX_LINK_JOBS="${{ inputs.maxLinkJobs || 4 }}" EXTRA_CMAKE_FLAGS="-DVELOX_ENABLE_ARROW=ON ${{ inputs.extraCMakeFlags }}"
+          ccache -s
+
+      - name: Upload presto fuzzer
+        uses: actions/upload-artifact@v3
+        with:
+          name: presto
+          path: velox/_build/debug/velox/expression/tests/velox_expression_fuzzer_test
+
+      - name: Upload spark fuzzer
+        uses: actions/upload-artifact@v3
+        with:
+          name: spark
+          path: |
+            velox/_build/debug/velox/expression/tests/spark_expression_fuzzer_test
+            velox/_build/debug/velox/expression/tests/spark_aggregation_fuzzer_test
+
+      - name: Upload aggregation fuzzer
+        uses: actions/upload-artifact@v3
+        with:
+          name: aggregation
+          path: velox/_build/debug/velox/exec/tests/velox_aggregation_fuzzer_test
+
+      - name: Upload join fuzzer
```

Comment:
Why do we need join fuzzer? Let's remove unnecessary steps.

Reply:
nit: These arent steps perse and arent run, this can be invoked when we are running join fuzzer to upload the fuzzer artifacts. I added these because I think we will be using exprimental for future experimental flags for all these jobs. However I will remove it since consensus is to only add when being used.

Reply:
for future reference you can disable steps by adding `if: false` to it, useful for testing and such ^^